### ISSUE TITLE
Implement functions for applying custom actions effects

### ISSUE THREAD

**Issue by brentd313** [uchicago-cs/chiventure#554]
I will be writing the implementations of the fundamental effect functions to be used with custom actions. The skeleton functions can currently be found in "atomic.c" in the custom-actions module of the "custom-actions/intial-headers" branch. I will be testing these functions as I implement them.

**Comment by brentd313** [comment]
I opened a new branch from dev, titled "custom-actions/effects", on which I have completed the implementation for the fundamental effect-applying functions for custom action execution.

**Comment by brentd313** [comment]
The implementation of the effect-applying functions can be found in the "custom-actions-effect.c" file in the src/custom-actions/src directory. Additionally, I have created a "custom-actions-effect.h" header and a "custom-actions-common.h" file in the src/custom-actions/include directory.

**Comment by brentd313** [comment]
There are two functions yet to be implemented: say_phrase() and move_player(). The implementation of these functions will be added once the necessary file destinations and data structures have been identified.

**Comment by brentd313** [comment]
I wrote unit tests for each of the effect-applying functions implemented in "custom-actions-effect.c". These tests can be found in the file "test_ca_effect.c" in the tests/custom-actions directory of the branch mentioned above.

**Comment by brentd313** [comment]
I created and updated the necessary CMake files in order to compile the implementations and tests. After addressing the errors raised by the build, the code is compiling error-free and all tests are passing.

**Comment by brentd313** [comment]
After integrating with the existing custom actions condition-checking code in dev, the pull request with the fundamental effect-applying functions has been merged into the dev branch, and the custom-actions/effects branch has been deleted. To review, these functions provide basic ways to apply effects that will define a custom action. In this issue, I implemented these functions and wrote unit tests. While these provide a good set of tools for applying the effects of an action, we will continue to update these implementations with improvements or new functions as need for new functionality presents itself. The say_phrase() and move_player() functions in "custom-actions-effect.h" provide good examples of what more advanced effect-applying functions might be for future implementation. The code referred to in this issue can be found either through the linked pull request or in the src/custom-actions and testing/custom-actions directories of the dev branch.
